util/fibers/listener_interface.cc

```diff
@@ -117,6 +118,14 @@ void ListenerInterface::RunAcceptLoop() {
 
     VSOCK(2, *peer) << "Accepted " << peer->RemoteEndpoint();
 
+    if (open_connections_ >= max_clients_) {
```

Comment:
I prefer explicit `open_connections_.load(memory_order_relaxed)` - you can also increment before with `uint32_t prev_count = open_connections_.fetch_add(1, std::memory_order_relaxed);` , compare it  and decrement if it crosses the limit

Reply:
good idea, thanks!